pkg/controllers/ingress_controller.go

```diff
@@ -211,14 +279,19 @@ func (r *IngressReconciler) newDnsEndpoint(ctx context.Context, dnsEndpoint *ext
 				RecordType:    "CNAME",
 				SetIdentifier: r.ClusterName,
 			}
-			r.addIngressTargetsToEndpoint(ep, ingress)
+			r.addIngressTargetsToEndpoint(ep, &ingress)
 
 			setEndpointProviderSpecificProperty(ep, WeightProperty, strconv.FormatUint(uint64(desiredWeight), 10))
```

Comment:
is not this set always being override by this one? https://github.com/adevinta/k8s-traffic-controller/pull/14/files#diff-8aff05a777f03ad6a8923481e32ba9482911d875f2fc4ad7007be613a281a544R239

Reply:
no.
There are 2 cases:
1. to inject DNS records pointing to the load balancer included in the ingresses
2. to inject DNS records pointing to the load balancer included in the services

This line is here to handle the ingress objects, line 239 is here to handle service objects